src/System.Xml.XmlSerializer/src/System/CodeDom/Compiler/CodeDomProvider.cs

```diff
@@ -40,131 +38,97 @@ public override string CreateEscapedIdentifier(string name)
-                "enum",
-                "goto",
-                "lock",
-                "long",
-                "null",
-                "this",
-                "true",
-                "uint",
-                "void",
-            },
-            new string[] {  // 5 characters
-                "break",
-                "catch",
-                "class",
-                "const",
-                "event",
-                "false",
-                "fixed",
-                "float",
-                "sbyte",
-                "short",
-                "throw",
-                "ulong",
-                "using",
-                "where",
-                "while",
-                "yield",
-            },
-            new string[] {  // 6 characters
-                "double",
-                "extern",
-                "object",
-                "params",
-                "public",
-                "return",
-                "sealed",
-                "sizeof",
-                "static",
-                "string",
-                "struct",
-                "switch",
-                "typeof",
-                "unsafe",
-                "ushort",
-            },
-            new string[] {  // 7 characters
-                "checked",
-                "decimal",
-                "default",
-                "finally",
-                "foreach",
-                "partial",
-                "private",
-                "virtual",
-            },
-            new string[] {  // 8 characters
-                "abstract",
-                "continue",
-                "delegate",
-                "explicit",
-                "implicit",
-                "internal",
-                "operator",
-                "override",
-                "readonly",
-                "volatile",
-            },
-            new string[] {  // 9 characters
-                "__arglist",
-                "__makeref",
-                "__reftype",
-                "interface",
-                "namespace",
-                "protected",
-                "unchecked",
-            },
-            new string[] {  // 10 characters
-                "__refvalue",
-                "stackalloc",
-            },
-        };
-
-        static CSharpCodeProvider()
+        private static bool IsKeyword(string value)
         {
-            s_fixedStringLookup = new Dictionary<string, object>();
-            for (int i = 0; i < s_keywords.Length; i++)
+            switch (value)
             {
-                string[] values = s_keywords[i];
-                if (values != null)
-                {
-                    for (int j = 0; j < values.Length; j++)
-                    {
-                        s_fixedStringLookup.Add(values[j], null);
-                    }
-                }
+                case "as": // 2 characters
+                case "do":
+                case "if":
+                case "in":
+                case "is":
+                case "for": // 3 characters
```

Comment:
The comment on string length isn't really relevant when it's no longer a trie keyed on length.